flake.nix

```
@@ -17,6 +17,21 @@
       nixpkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
       nixpkgsFor-bosh-cli-v7-3-1 = forAllSystems (system: import nixpkgs-bosh-cli-v7-3-1 { inherit system; });
     in {
+      packages = forAllSystems (system:{
+        app-autoscaler-cli-plugin = nixpkgsFor.${system}.buildGoModule rec {
+          pname = "app-autoscaler-cli-plugin";
+          version = "f46dc1ea62c4c7bd426c82f4e2a525b3a3c42300";
+          src = nixpkgsFor.${system}.fetchgit {
+            url = "https://github.com/cloudfoundry/app-autoscaler-cli-plugin";
+            rev = "${version}";
```

That's probably not the version we want to get. :)
Sometimes there are problems, when you specify both, `ref` and `rev`. Maybe you have run into those troubles.
```suggestion
          version = "3.0.0";
          src = nixpkgsFor.${system}.fetchgit {
            url = "https://github.com/cloudfoundry/app-autoscaler-cli-plugin";
            ref = "refs/tags/v${version}"; # Comment me out, if I don't work.
            rev = "ec170880a14d21134332d70aaf7dea3ba3c9364a"; 
```

---

The point that `version` contains a git-rev-spec instead of a version, is still valid.
```suggestion
          # version = "> 3.0.0";
          src = nixpkgsFor.${system}.fetchgit {
            url = "https://github.com/cloudfoundry/app-autoscaler-cli-plugin";
            rev = "f46dc1ea62c4c7bd426c82f4e2a525b3a3c42300";
```

---

Why isn't it a good idea to simply use the commit-hash as `version`?  🤔

---

To distinguish between what is commonly considered as version (when depending on a ruby-gem, a python-package, an aws-cli, …) and what is a git-revision.

Why confuse the consumer with that strange hash being shown as the version? How can he compare version-numbers to judge if it is newest or not?

On the other side there is the “advantage” that he – at least – can later easily by the package-name follow back to the actual code without having to look at the nix-definition.

---

That's fine for me.